crypto/pkcs12/p12_mutl.c

```diff
@@ -237,7 +258,8 @@ int PKCS12_setup_mac(PKCS12 *p12, int iter, unsigned char *salt, int saltlen,
     }
     p12->mac->salt->length = saltlen;
     if (!salt) {
-        if (RAND_bytes(p12->mac->salt->data, saltlen) <= 0)
+        if (RAND_bytes_ex(p12->authsafes->ctx.libctx, p12->mac->salt->data,
```

Comment:
is authsafes ever NULL here?

Reply:
It's allocated in the ASN1 auto-generated PKCS12_new() function so I don't think so...